heron/stmgr/src/cpp/manager/stmgr-client.cpp

```diff
@@ -160,7 +163,16 @@ void StMgrClient::HandleHelloResponse(void*, proto::stmgr::StrMgrHelloResponse*
   client_manager_->HandleStMgrClientRegistered();
 }
 
-void StMgrClient::OnReConnectTimer() { Start(); }
+void StMgrClient::OnReConnectTimer() {
+  reconnect_attempts_ += 1;
```

Comment:
If stmgr A lost connection to stmgr B and tried 28 times on Monday, and re-connected successfully at the 29th attempt. Four days later on Friday, stmgr A lost connection to stmgr B again and tried to re-connect 2 times and quit.
I suggest resetting `reconnect_attempts_ `to 0 after A resumed connection to B on Monday.